src/sentry/incidents/logic.py

```diff
@@ -555,6 +555,30 @@ def create_alert_rule(
     return alert_rule
 
 
+def snapshot_alert_rule(alert_rule):
+    # Creates an archived alert_rule using the same properties as the passed rule
+    # It will also resolve any incidents attached to this rule.
+    triggers = AlertRuleTrigger.objects.filter(alert_rule=alert_rule)
+
+    incidents = Incident.objects.filter(alert_rule=alert_rule)
+
+    alert_rule.id = None
```

Comment:
I wonder if we can clone the alert rule instance in some way before we modify it. It's a little weird that the snapshot function modifies it. As far as I know it's ok to `deepcopy` an instance